modules/nlu/src/backend/api.ts

```diff
@@ -15,28 +20,58 @@ export const PredictSchema = Joi.object().keys({
   text: Joi.string().required()
 })
 
-export default async (bp: typeof sdk, state: NLUState) => {
-  const router = bp.http.createRouterForBot('nlu')
+const makeErrorMapper = (bp: typeof sdk) => (err: { botId: string; lang: string; error: Error }, res: Response) => {
+  const { error, botId, lang } = err
+
+  if (error instanceof BotNotMountedError) {
+    return res.status(404).send(`Bot ${botId} doesn't exist`)
+  }
+
+  if (error instanceof BotDoesntSpeakLanguageError) {
+    return res.status(422).send(`Language ${lang} is either not supported by bot or by language server`)
+  }
+
+  const msg = 'An unexpected error occured.'
+  bp.logger
+    .forBot(botId)
+    .attachError(error)
+    .error(msg)
+  return res.status(500).send(msg)
+}
+
+const mapTrainSession = (ts: TrainingSession): sdk.NLU.TrainingSession => {
+  const { botId, language, progress, status } = ts
+  const key = `training:${botId}:${language}`
+  return { key, language, status, progress }
+}
+
+export const getWebsocket = (bp: typeof sdk) => {
```

Comment:
this naming is missLeading .. it doesn't return an actual websocket (socket.io/Socket) and the return type isn't defined. If you just give this a proper name  it should be fine.